diofant/integrals/tests/test_meijerint.py

```diff
@@ -651,3 +651,8 @@ def test_issue_6860():
 def test_issue_8368():
     assert meijerint_indefinite(cosh(x)*exp(-x*t), x) == (
         (-t - 1)*exp(x) + (-t + 1)*exp(-x))*exp(-t*x)/2/(t**2 - 1)
+
+
+def test_issue_4311():
+    assert meijerint_indefinite(x*abs(9 - x**2), x) == \
+        -81*meijerg(((3, 1), (2,)), ((1,), (2, 0)), x**2/9)
```

Comment:
On Sun, Jul 31, 2016 at 08:23:20AM -0700, Kalevi Suominen wrote:
>    Thanks. I'm sure I'll be interested in any progress you make. I have been
>    thinking about the odd minus sign.

But keep in mind, if you look on the numerical testing above - it seems that
exact solution (produced by using Piecewise) and meijerint_indefinite's
answer differ not just in sign.